packages/mdc-grid-list/_mixins.scss

```diff
@@ -0,0 +1,251 @@
+  $feat-color: mdc-feature-create-target($query, color);
+  $feat-structure: mdc-feature-create-target($query, structure);
+  $feat-typography: mdc-feature-create-target($query, typography);
+
+  // postcss-bem-linter: define grid-list
+  @include mdc-grid-list-tile-aspect(1, $query: $query);
+  @include mdc-grid-list-tile-gutter(4px, $query: $query);
+
+  .mdc-grid-list__tiles {
+    @include mdc-feature-targets($feat-structure) {
+      display: flex;
+      flex-flow: row wrap;
+      padding: 0;
+    }
+  }
+
+  .mdc-grid-list--tile-gutter-1 {
+    @include mdc-grid-list-tile-gutter(1px, $query: $query);
+  }
+
+  .mdc-grid-list--tile-aspect-16x9 {
+    @include mdc-grid-list-tile-aspect(16 / 9, $query: $query);
+  }
+
+  .mdc-grid-list--tile-aspect-3x2 {
+    @include mdc-grid-list-tile-aspect(3 / 2, $query: $query);
+  }
+
+  .mdc-grid-list--tile-aspect-2x3 {
+    @include mdc-grid-list-tile-aspect(2 / 3, $query: $query);
+  }
+
+  .mdc-grid-list--tile-aspect-4x3 {
+    @include mdc-grid-list-tile-aspect(4 / 3, $query: $query);
+  }
+
+  .mdc-grid-list--tile-aspect-3x4 {
+    @include mdc-grid-list-tile-aspect(3 / 4, $query: $query);
+  }
+  // postcss-bem-linter: end
+
+  // postcss-bem-linter: define grid-tile
+  .mdc-grid-tile {
+    @include mdc-feature-targets($feat-structure) {
+      display: block;
+      position: relative;
+      /* @alternate */
+      width: $mdc-grid-list-tile-width;
+      width: var(--mdc-grid-list-tile-width, $mdc-grid-list-tile-width);
+    }
+  }
+
+  .mdc-grid-tile__primary {
+    @include mdc-feature-targets($feat-color) {
+      @include mdc-theme-prop(background-color, background);
+      @include mdc-theme-prop(color, text-primary-on-background);
+    }
+
+    @include mdc-feature-targets($feat-structure) {
+      position: relative;
+      height: 0;
+    }
+  }
+
+  .mdc-grid-tile__primary-content {
+    @include mdc-feature-targets($feat-structure) {
+      position: absolute;
+      top: 0;
+      right: 0;
+      bottom: 0;
+      left: 0;
+      width: 100%;
+      height: 100%;
+      background-repeat: no-repeat;
+      background-position: center;
+      background-size: cover;
+    }
+  }
+
+  .mdc-grid-tile__secondary {
+    @include mdc-feature-targets($feat-color) {
+      @include mdc-theme-prop(background-color, primary);
+      @include mdc-theme-prop(color, on-primary);
+    }
+
+    @include mdc-feature-targets($feat-structure) {
+      position: absolute;
+      bottom: 0;
+      box-sizing: border-box;
+      height: $mdc-grid-list-oneline-cap-secondary-height;
+      padding: $mdc-grid-list-tile-secondary-padding;
+    }
+  }
+
+  .mdc-grid-tile__title {
+    @include mdc-typography-overflow-ellipsis($query: $query);
+
+    @include mdc-feature-targets($feat-structure) {
+      display: block;
+      margin: 0;
+      padding: 0;
+      font-size: 1rem;
+      line-height: 1rem;
+    }
+
+    @include mdc-feature-targets($feat-typography) {
```

Comment:
Curious what does the typography includes? Should font-size and line-height be part of typography?

Reply:
It's primarily intended to include things that depend on the mdc-typography scale variables. Thinking on this particular line again (which @mmalerba brought up), I'm not even sure it's absolutely necessary for this, since this is just font-weight... Thoughts, Miles? What's the rule of thumb for things falling under the typography feature aside from direct uses of the `mdc-typography` mixin?

Reply:
Like you said, I think if it depends on variables that we want the user to configure then it should be in the typography feature. That way users can do something like this:

```sass
@include mdc-grid-list-core-styles(mdc-without(typography));

.normal-typography-section {
  @include mdc-grid-list-core-styles(typography);
}

.fancy-typography-section {
  $mdc-typography-font-weight-values: $fancy-typography-weights !global;
  @include mdc-grid-list-core-styles(typography);
}
```

Basically when they call the mixin with the `typography` feature as the query, they're asking "give me all of the style rules that might have changed as a result of my messing with the typography variables"